torch/_subclasses/fake_tensor.py

```diff
@@ -77,75 +162,9 @@ def __repr__(self):
-            assert len(args) == 1 and isinstance(args[0], FakeTensor)
-            return args[0].fake_device
-
-        def wrap(e, device=None):
-            if isinstance(e, torch.Tensor) and not isinstance(e, FakeTensor):
-                if device:
-                    return FakeTensor(e, device)
-                else:
-                    return FakeTensor.from_tensor(e)
-            else:
-                return e
-
-        # if we are in the dispatch mode, we will enter this function even if the inputs
-        # are not FakeTensors, and they need to be wrapped
-        if cls == FakeTensorMode:
-            args = tree_map(wrap, args)
-            kwargs = tree_map(wrap, kwargs)
-
-        # _to_copy fails when run with FakeTensors to cuda device
-        # TODO: debug
-        if func == torch.ops.aten._to_copy.default:
-            _, new_kwargs = normalize_function(
-                func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
-            )
-            out_device = new_kwargs.pop("device", new_kwargs["input"].device)
-            with no_dispatch():
-                input = new_kwargs.pop("input").to("meta")
-                return FakeTensor(
-                    torch.ops.aten._to_copy(input, **new_kwargs), out_device
-                )
-
-        if _is_tensor_constructor(func):
-            assert func not in _non_kwarg_device_constructors
-            _, new_kwargs = normalize_function(
-                func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
-            )
-            # cpu is default device if none is specified
-            out_device = new_kwargs.pop("device", torch.device("cpu"))
-            new_kwargs["device"] = torch.device("meta")
-            r = super().__torch_dispatch__(func, types, (), new_kwargs)
-            return FakeTensor(r, out_device)
-
-        r = super().__torch_dispatch__(func, types, args, kwargs)
-
-        # TODO: handle non-kwarg devices
-        assert func not in _device_not_kwarg_ops, f"NYI: {func}"
-
-        # if device is specified, use that
-        if kwargs.get("device", None):
-            return tree_map(partial(wrap, device=kwargs["device"]), r)
-
-        # operators which copy size from another tensor do not
-        # also take device from the size tensor
-        # other size_as operators are not builtin operators
-        if func == aten.resize_as_.default:
-            _, new_kwargs = normalize_function(
-                func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
-            )
-            # device of the input is returned
-            return tree_map(partial(wrap, device=new_kwargs["input"].device), r)
-
-        common_device = FakeTensor._find_common_device(func, args, kwargs)
-
-        return tree_map(partial(wrap, device=common_device), r)
+        def run_fn(func, types, args, kwargs):
+            return torch.Tensor.__torch_dispatch__(func, types, args, kwargs)
```

Comment:
We should be okay running `func(*args, **kwargs)` in all cases. This way we can also remove the `run_function` argument

Reply:
This actually causes an error: 
```
x = FakeTensor.from_tensor(torch.tensor(0.0))
y = FakeTensor.from_tensor(torch.rand([4, 4], device="cuda"))
out = x + y
...
ERROR: test_zero_dim (__main__.FakeTensorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_fake_tensor.py", line 32, in test_zero_dim
    out = x + y
TypeError: unsupported operand type(s) for +: 'FakeTensor' and 'FakeTensor'
```

Although it's a mute point because I'm going to do the refactoring of tensors storing mode and calling into it